Adding chia peer command

wallentx opened the pull request:
Addition of chia peer command for (wallet, farmer, full node and harvester) & slight refactor of chia show.

wallentx committed:
Adding chia peer command

wallentx committed:
Cleaning up peer and show commands

wallentx committed:
Formatting to please flake8

altendky commented on chia/cmds/peer.py:
```diff
@@ -0,0 +1,157 @@
+from chia.util.network import is_trusted_inner
+
+
+async def print_connections(client, time, NodeType, trusted_peers: Dict):
```
If you need the `time` module, go ahead and import it.

```suggestion
async def print_connections(client, NodeType, trusted_peers: Dict) -> None:
```

altendky commented on chia/cmds/peer.py:
```diff
@@ -0,0 +1,157 @@
+        client = await FullNodeRpcClient.create(self_hostname, uint16(rpc_port), DEFAULT_ROOT_PATH, config)
+
+        if show_connections:
+            trusted_peers: Dict = config["full_node"].get("trusted_peers", {})
```
I think this is what comes out of the config?

```suggestion
            trusted_peers: Dict[str, str] = config["full_node"].get("trusted_peers", {})
```

wallentx committed:
Update chia/cmds/peer.py

Adding as suggested

Co-authored-by: Kyle Altendorf <sda@fstab.net>

wallentx committed:
Making suggested changes to peer.py

wallentx committed:
Merge branch 'wallentx/chia-peer-command' of github.com:Chia-Network/chia-blockchain into wallentx/chia-peer-command

Commit:
Merge branch 'main' into wallentx/chia-peer-command

Commit:
finalize pr

very nice
i basically did half the work when i refactored it

Co-Authored-By: William Allen <william.allentx@gmail.com>

jack60612 commented:
Testing locally now

jack60612 commented:
Local CLI tests successful!!

Commit:
fix lint, damn isort

REEEE

Co-Authored-By: William Allen <william.allentx@gmail.com>

jack60612 commented:
> This pull request **introduces 8 alerts** when merging [d398434](https://github.com/Chia-Network/chia-blockchain/commit/d3984347973b3989a1827e735bbe59b189bdd7d3) into [6923ad8](https://github.com/Chia-Network/chia-blockchain/commit/6923ad8e12a376e7f5fa428a0cea0de70ce53e7f) - [view on LGTM.com](https://lgtm.com/projects/g/Chia-Network/chia-blockchain/rev/pr-d254e07bc975b818c0b621089bdf3fb6b4bf2367)
> 
> **new alerts:**
> 
> * 8 for Module-level cyclic import

False alarm

Commit:
allow passthrough of root_dir

Commit:
yikes

emlowe commented:
This break backwards compatibility right - `chia show -c` doesn't work anymore?  Should we maintain the `chia show` options? At the very least I think `chia show -c` should print that it is deprecated and use `chia peer -c`?

jack60612 commented:
> This break backwards compatibility right - `chia show -c` doesn't work anymore? Should we maintain the `chia show` options? At the very least I think `chia show -c` should print that it is deprecated and use `chia peer -c`?

i can add that. It does break backwards compatibility

Commit:
Update show.py

emlowe commented on chia/cmds/show.py:
```diff
@@ -229,19 +229,24 @@ async def show_async(
 @click.option("-s", "--state", help="Show the current state of the blockchain", is_flag=True, type=bool, default=False)
 @click.option(
-    "-bh", "--block-header-hash-by-height", help="Look up a block header hash by block height", type=str, default=""
+    "-h", "--block-header-hash-by-height", help="Look up a block header hash by block height", type=str, default=""
```
breaking this backwards compat should be avoided (I understand the multi-letter options are a problem, but we shouldn't just break things)

Commit:
fix

emlowe commented:
You also need to fix `chia show -a` and `chia show -r` and anythign else removed from `chia show`

For example, I know the `--wallet-rpc-port` is unused, but by removing it you break scripts that were using it accidentally - it would be better to leave it in and make the help say it's unused and deprecated. Oh, click has a deprecated flag https://click.palletsprojects.com/en/8.1.x/api/?highlight=deprecated#click.Command

jack60612 commented:
> You also need to fix `chia show -a` and `chia show -r` and anythign else removed from `chia show`
> 
> For example, I know the `--wallet-rpc-port` is unused, but by removing it you break scripts that were using it accidentally - it would be better to leave it in and make the help say it's unused and deprecated. Oh, click has a deprecated flag https://click.palletsprojects.com/en/8.1.x/api/?highlight=deprecated#click.Command

ok that makes sense.

Commit:
happiness is a thing

jack60612 commented:
@emlowe is it good now?

Commit:
Update show.py

jack60612 commented:
all fixed

emlowe commented on chia/cmds/show.py:
```diff
@@ -326,16 +217,6 @@ async def show_async(
-    ),
-    type=int,
-    default=None,
-)
```
This needs to be left in - it's at least used later on.

jack60612 replied:
This was accidently removed, i'm going to re review this whole pr before making any more changes

jack60612 replied:
Sorry about the review spam, should be good now

Commit:
re add rpc, oops

mariano54 reviewed (changes requested):
I think the `chia peer` command should support any service (farmer, harvester, wallet etc), since they all support the same RPC calls.

This would make it more generic and useful for all types of situations. `chia show` was meant to be just for the node, but if we are refactoring this RPC, I think we should have a higher standard and make it generalizeable.

Commit:
allow any node type, per request

Commit:
make separate funcs files to align with rest of codebase

Commit:
Revert "make separate funcs files to align with rest of codebase"

This reverts commit dad21be089b7efe383cf047710d02ea260c84168.

Commit:
Merge branch 'main' into wallentx/chia-peer-command

Commit:
remake changes

Commit:
lint & finish chia peer

Commit:
isort

Commit:
lint

Commit:
tad more lint never hurt

Commit:
cleaner code  is fun

Kyle saved me from type hell

Co-Authored-By: Kyle Altendorf <sda@fstab.net>

Commit:
final change

oops, last lint

mypy ignore

isort

Revert "isort"

This reverts commit 4bef38a885c0481ec844a9f06185a4ef993f7374.

final change

isort does not work on windows

Commit:
Update cmds_util.py

Commit:
Merge branch 'main' into wallentx/chia-peer-command

Commit:
Update chia.py

jack60612 commented:
both the chia peer and the chia show command were successfully tested locally.

Commit:
remove accidental  change

mariano54 commented on chia/cmds/cmds_util.py:
```diff
@@ -10,3 +26,48 @@ def transaction_submitted_msg(tx: TransactionRecord) -> str:
+    return True
+
+
+async def execute_with_any_node(
```
I would rename to `execute_with_any_service`. Node means fulls node in chia vocabulary.

jack60612 replied:
Renamed!

mariano54 reviewed (dismissed):
LGTM apart from small nit for naming

Commit:
rename function

makes sense to me

Commit:
Merge branch 'main' into wallentx/chia-peer-command

mariano54 reviewed (changes requested):
Need to fix the LGTM issues (cyclic import)

jack60612 commented:
> Need to fix the LGTM issues (cyclic import)

they are wrong, i dont know what they are yapping about.

emlowe commented:
Yea, LGTM has some issues - Kyle would actually like us to stop using it as it's not adding much value

mariano54 commented:
We can't merge this PR if it adds 40 LGTM issues. 
Or should we just start ignoring it completely?

jack60612 commented:
closing this pr in favor of #12532
